os/kernel/binary_manager/binary_manager_load.c

```diff
@@ -597,22 +605,64 @@ static int reloading_thread(int argc, char *argv[])
 static int update_thread(int argc, char *argv[])
 {
 	int ret;
+	int bin_idx;
+	uint32_t bin_count = binary_manager_get_ucount();
 
 	/* Check whether there is kernel binary for update.
 	 * If kernel update exists, board will be rebooted. Or, it will return negative values. */
 	ret = binary_manager_update_kernel_binary();
-	if (ret != BINMGR_ALREADY_UPDATED) {
-		/* Return errors except for BINMGR_ALREADY_UPDATE which means already the latest kernel binary is running */
+	if (ret != BINMGR_ALREADY_UPDATED && ret != BINMGR_NOT_FOUND) {
+		/* Return errors except for BINMGR_ALREADY_UPDATE which means already the latest kernel binary is running
+		 * and BINMGR_NOT_FOUND means there is no valid binary in inactive partition
+		 */
 		return ret;
 	}
 
+#ifndef CONFIG_USE_BP
+	/* In case of using BP, checking new binaries and updating BP is done when calling binary_manager_set_bootparam.
+	 * When not using BP, check if new common binary or user binaries exist at this time.
+	*/
+#ifdef CONFIG_APP_BINARY_SEPARATION
+	bool need_update = false;
+
+	/* Reload binaries if new binary is scanned */
+	for (bin_idx = 1; bin_idx <= bin_count; bin_idx++) {
```

Comment:
hmm I think code size can be reduced, please consider below.
```
#ifdef CONFIG_APP_BINARY_SEPARATION
#ifdef CONFIG_SUPPORT_COMMON_BINARY
	bin_idx = BM_CMNLIB_IDX; //check from BM_CMNLIB_IDX if common binary supported
#else
	bin_idx = 1;
#endif
	while (bin_idx <= bin_count) {
		/* Scan binary files */
		ret = binary_manager_check_user_update(bin_idx);
		if (ret == BINMGR_OK) {
			/* Update index for inactive partition */
			update_bp_data.app_data[BIN_BPIDX(bin_idx)].useidx ^= 1;
			need_update = true;
		} else {
			bmdbg("Fail to check update, idx : %d ret : %d\n", bin_idx, ret);
		}
		bin_idx++;
	}

	if (!need_update) {
		bmdbg("No App binaries to update\n");
		return BINMGR_ALREADY_UPDATED;
	}
#endif /* End of CONFIG_APP_BINARY_SEPARATION */
```

Reply:
Modified.